.github/workflows/test-e2e-gov.yml

```diff
@@ -13,6 +13,7 @@ jobs:
       - name: Check out code
         uses: actions/checkout@v4
         with:
+          ref: ${{github.event.pull_request.head.sha}}
```

Comment:
This might be my fault. We were fighting some strange issue with PR target events, and I think I touched more places than needed, like this one. We have removed the PR target issue now, on a separate PR I merged before.

I think we can remove this, but I let it to @roothorp for the final decision.